weldx/transformations.py

```diff
@@ -379,45 +518,449 @@ def _calculate_orthogonal_axis(a_0, a_1):
+        length_path = len(path) - 1
+        if length_path > 1:
+            for i in np.arange(1, length_path):
+                lcs = lcs + self.graph.edges[path[i], path[i + 1]]["lcs"]
+        return lcs
+
+    def has_coordinate_system(self, coordinate_system_name: Hashable) -> bool:
+        """
+        Return 'True' if a coordinate system with specified name is part of the class.
+
+        :param coordinate_system_name: Name of the coordinate system, that should be
+        checked.
+        :return: 'True' or 'False'
+        """
+        return coordinate_system_name in self._graph.nodes
+
+    def has_data(self, coordinate_system_name: Hashable, data_name: Hashable) -> bool:
+        """
+        Return 'True' if the desired coordinate system owns the specified data.
+
+        :param coordinate_system_name: Name of the coordinate system
+        :param data_name: Name of the data
+        :return: 'True' or 'False'
+        """
+        return data_name in self._graph.nodes[coordinate_system_name]["data"]
+
+    def get_data(self, data_name, target_coordinate_system_name=None):
+        """
+        Get the specified data, optionally transformed into any coordinate system.
+
+        :param data_name: Name of the data
+        :param target_coordinate_system_name: Name of the target coordinate system. If
+        it is not None or not identical to the owning coordinate system name, the data
+        will be transformed to the desired system.
+        :return: (Transformed) data
+        """
+        data_struct = self._data[data_name]
+        if (
+            target_coordinate_system_name is None
+            or target_coordinate_system_name == data_struct.coordinate_system_name
+        ):
+            return data_struct.data
+        else:
+            return self.transform_data(
+                data_struct.data,
+                data_struct.coordinate_system_name,
+                target_coordinate_system_name,
+            )
+
+    def transform_data(
+        self,
+        data: Union[xr.DataArray, np.ndarray, List],
+        source_coordinate_system_name: Hashable,
+        target_coordinate_system_name: Hashable,
+    ):
+        """
+        Transform spatial data from one coordinate system to another.
+
+        :param data:
+        :param source_coordinate_system_name: Name of the coordinate system the data is
+        defined in
+        :param target_coordinate_system_name: Name of the coordinate system the data
+        should be transformed to
+        :return: Transformed data
+        """
+        lcs = self.get_local_coordinate_system(
+            source_coordinate_system_name, target_coordinate_system_name
+        )
+        if isinstance(data, xr.DataArray):
+            mul = ut.xr_matmul(
+                lcs.orientation,
+                data,
+                dims_a=["c", "v"],
+                dims_b=["c", "n"],
+                dims_out=["c", "n"],
+            )
```

Comment:
Since this is matrix * vector it should look like this (without specifying __n__) for the data dimensions
```python
mul = ut.xr_matmul(
    lcs.orientation,
    data,
    dims_a=["c", "v"],
    dims_b=["c"],
    dims_out=["c"],
)
```
However, output will be of shape (n,3) instead of (3,n). I think we should agree on (n,3)-like shapes by default for broadcasting.